pkg/identity/ciprovider/principal.go

```diff
@@ -66,7 +66,7 @@ func applyTemplateOrReplace(extValueTemplate string, tokenClaims map[string]stri
 		var doc bytes.Buffer
 		// This option forces to having the claim that is required
 		// for the template
-		t := template.New("").Option("missingkey=error")
+		t := template.New("").Option("missingkey=zero")
```

Comment:
Pinged offline about this, but WDYT about what you had mentioned originally, changing the behavior of `default-key-template` to be a default? Then we can maintain the logic of throwing an error if a token is malformed.

Reply:
I can do that. on it

Reply:
Done